patches/amd-mainline/rocThrust/0001-Forward-CMAKE_BUILD_TYPE-when-building-gtest.patch

```diff
@@ -0,0 +1,26 @@
+From 3d3bcc3540e3fc09a94483134bbe8813f6fa7495 Mon Sep 17 00:00:00 2001
+From: Scott <scott.todd0@gmail.com>
+Date: Mon, 7 Apr 2025 15:01:18 -0700
+Subject: [PATCH] Forward CMAKE_BUILD_TYPE when building gtest.
+
+This avoids `lld-link: error: /failifmismatch: mismatch detected for '_ITERATOR_DEBUG_LEVEL':` errors on Windows when rocThrust is compiled in Release instead of Debug (the default).
+---
+ cmake/Dependencies.cmake | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/cmake/Dependencies.cmake b/cmake/Dependencies.cmake
+index b0bd252f..2088312f 100644
+--- a/cmake/Dependencies.cmake
++++ b/cmake/Dependencies.cmake
+@@ -53,7 +53,7 @@ if(BUILD_TEST OR BUILD_HIPSTDPAR_TEST)
+       GIT_REPOSITORY      https://github.com/google/googletest.git
+       GIT_TAG             release-1.11.0
+       INSTALL_DIR         ${GTEST_ROOT}
+-      CMAKE_ARGS          -DBUILD_GTEST=ON -DINSTALL_GTEST=ON -Dgtest_force_shared_crt=ON -DBUILD_SHARED_LIBS=OFF -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
++      CMAKE_ARGS          -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} -DBUILD_GTEST=ON -DINSTALL_GTEST=ON -Dgtest_force_shared_crt=ON -DBUILD_SHARED_LIBS=OFF -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
```

Comment:
Note that rocThrust already adds this CMake option for `googlebenchmark`: https://github.com/ROCm/rocThrust/blob/dfb07406c7ecc401e82a05497f99b0ef5aeb13f8/cmake/Dependencies.cmake#L167

```cmake
    download_project(
      PROJ                googlebenchmark
      GIT_REPOSITORY      https://github.com/google/benchmark.git
      GIT_TAG             v${BENCHMARK_VERSION}
      GIT_SHALLOW         TRUE
      INSTALL_DIR         ${GOOGLEBENCHMARK_ROOT}
      CMAKE_ARGS          -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} -DBUILD_SHARED_LIBS=OFF -DBENCHMARK_ENABLE_TESTING=OFF -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR> -DCMAKE_CXX_STANDARD=14 ${COMPILER_OVERRIDE}
```